### PR TITLE
Rewrite comparison.md in third-person

### LINKED ISSUE #407: comparison.md is written in first-person
As reported by @jbduncan in #405 (https://github.com/google/truth/pull/405/files/9191f6092da5594c69b3204c31b37ca32213cebd..edcb1037822d5fa2348e31ed09c5524facb88900#diff-d542fbe4e3828c953182a69002371df8) the `comparison.md` doc is written in first person - we may want to consider rewriting it in third-person.

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #407
@ronshapiro /cc
@jbduncan /cc

### CODE REVIEW

File: comparison.md [hunk 1]
```diff
@@ -15,15 +15,14 @@ from [FEST]. Truth is significantly different from [Hamcrest], which uses
 ## Why create Truth when AssertJ already exists?
 
 In other words: Maybe Truth is better for some use cases, but why not just
-improve AssertJ instead? The reason is historical: AssertJ didn't exist when we
-started Truth. By the time it was created, we'd begun using Truth widely
-internally, and we'd added some features that would be difficult to retrofit
+improve AssertJ instead? The reason is historical: AssertJ didn't exist when
+Truth was started. By the time it was created, Truth was already widely used
```
Comment (ronshapiro) [inline comparison.md]: Maybe "when Truth was created"?

File: comparison.md [hunk 2]
```diff
@@ -15,15 +15,14 @@ from [FEST]. Truth is significantly different from [Hamcrest], which uses
...
+internally, and some features were added that would be difficult to retrofit
```
Comment (ronshapiro) [inline comparison.md]: Instead of "internally", "throughout Google"?

File: comparison.md [hunk 3]
```diff
@@ -15,15 +15,14 @@ from [FEST]. Truth is significantly different from [Hamcrest], which uses
...
 onto AssertJ.
 
 ## Comparison vs. Hamcrest
 
-Because Truth and Hamcrest differ so significantly, I'll cover only the main
-points:
+Truth and Hamcrest differ significantly, but here are the main differences:
```
Comment (ronshapiro) [inline comparison.md]: Truth and Hamcrest differ significantly. Their main differences include:

File: comparison.md [hunk 4]
```diff
@@ -65,9 +63,9 @@ also mean more convenience and power.
 
 ### Failure messages
 
-I think that AssertJ generally does this better at the moment. It follows an
+Probably AssertJ is generally doing this better at the moment. It follows an
```
Comment (ronshapiro) [inline comparison.md]: I imagine @cpovirk may have some comments on updating this

File: comparison.md [hunk 5]
```diff
@@ -98,8 +96,7 @@ assertThat(multimap).valuesForKey(x).containsExactly(y, z);
 ```
 
 Our philosophy has been that it's clearer to support only one kind of chaining,
```
Comment (ronshapiro) [inline comparison.md]: "Truth's philosophy has been..."

Queue all of the capital-T truth jokes ;)

File: comparison.md [hunk 6]
```diff
@@ -98,8 +96,7 @@ assertThat(multimap).valuesForKey(x).containsExactly(y, z);
...
-but I suspect that the AssertJ style is generally clear, too, and it can be
-convenient.
+but the AssertJ style is supposedly generally clear and can be convenient.
```
Comment (ronshapiro) [inline comparison.md]: I wonder how much this sentence really adds. We may want to just remove it.

File: comparison.md [hunk 7]
```diff
@@ -109,19 +106,19 @@ AssertJ provides assertions for several libraries' types. As of this writing,
 Truth includes assertions for [Guava] and [Protocol Buffers].
 
 Both have third-party extensions, such as for Android types
-([AssertJ][AssertJ-Android], [Truth][Truth-Android]). I don't have a feel for
-the overall size of each ecosystem.
+([AssertJ][AssertJ-Android], [Truth][Truth-Android]).
+It's unclear how large each ecosystem is overall.
```
Comment (ronshapiro) [inline comparison.md]: This may also have changed since when Chris first published this.

File: comparison.md [hunk 8]
```diff
@@ -109,19 +106,19 @@ AssertJ provides assertions for several libraries' types. As of this writing,
...
 
 ### Platform support (Android, GWT)
 
 AssertJ supports Android (though I had to use 2.x because the dexer rejected
```
Comment (ronshapiro) [inline comparison.md]: This is still in first person, but it seems like unrelated details by now? @cpovirk?

File: comparison.md [hunk 9]
```diff
@@ -109,19 +106,19 @@ AssertJ provides assertions for several libraries' types. As of this writing,
...
-3.x, even when I used only `Java6Assertions`). Possibly this was an issue with
-my build setup.
+3.x, even when using only `Java6Assertions`). Possibly this was an issue with
+the build setup.
 
 Truth supports Android in all its versions. The downside is that it requires you
 to look in a separate class for Java 8 assertions.
 
-Truth also supports [GWT], but we haven't made the necessary artifacts available
+Truth also supports [GWT], but the necessary artifacts haven't been made available
```
Comment (ronshapiro) [inline comparison.md]: I think this is no longer true? @cpovirk?

**Comment by cpovirk** [comment]
I think it's reasonable to standardize the mix of "I" and "we" that we currently have, but I think that switching to passive tense is a step backward for readability. I understand that many people don't like "I" and "we" in technical writing, but [different people feel differently about this](https://twp.duke.edu/sites/twp.duke.edu/files/file-attachments/first-person.original.pdf). Since there's disagreement, I'd rather err on the side of being clear.

**Comment by jbduncan** [comment]
I think @cpovirk makes a reasonable point. I was going to say originally that we stick to the third-person, but I think now that standardising on "we" and "we at Google" would be a good first step to undertake. If upon reading specific parts of the documentation we collectively realise that using "we" doesn't work so well, we can then use the third-person.

File: comparison.md (hunk 7 above)
Comment (cpovirk) [reply comparison.md]: I still haven't tried to measure it. Maybe we'll do so someday.

File: comparison.md (hunk 8 above)
Comment (cpovirk) [reply comparison.md]: The right thing for me to do here is to try 3.0 with some other build system and see if it works, then file a bug against AssertJ about the problems. In the meantime, I want to point out that AssertJ might not actually work with Android in practice, despite its best attempts, but that I'm not sure how widespread the problem is.

File: comparison.md (hunk 9 above)
Comment (cpovirk) [reply comparison.md]: Oops, you're right. This was not true even when I wrote this! I forgot that we generate `truth-gwt` artifact without using a separate project like we do for Guava.

**Comment by cpovirk** [comment]
> I feel bad that we led you to rewrite this by our filing a bug about it. But I continue to think that we're better off leaving the first-person phrasing intact. Sorry about that.

And in particular, I had a bunch of time to post my overall views on the bug before you put together the pull request :(

**Comment by Androbin** [comment]
Okay, feel free to continue from here ;)
